### ISSUE TITLE
Pseudorandom number generation in taobench slowpaths

### ISSUE THREAD

**Issue by ss7pro** [facebookresearch/DCPerf#21]
Folks,

Is it possible to replace taobench pseudo-random number generation in [taobench](https://github.com/facebookresearch/DCPerf/blob/a0a59e80091e2e8bb84b402cda7bdb44aff18379/packages/tao_bench/tao_bench_memcached_0001.diff#L204) with implementation we have in [feedsim](https://github.com/facebookresearch/DCPerf/blob/a0a59e80091e2e8bb84b402cda7bdb44aff18379/packages/feedsim/third_party/src/workloads/ranking/generators/RankingGenerators.h#L31)?

**Comment by lukalt** [comment]
> Does `folly::Random::rand32()` has the same lock contention problem as in `std::rand()`?

No, it uses a thread-local random generator, so no global lock here.
